packages/lib/shared/components/btns/button-group/ButtonGroup.tsx

```diff
@@ -1,66 +1,141 @@
+  Popover,
+  PopoverContent,
+  PopoverTrigger,
+  Text,
+} from '@chakra-ui/react'
+import { LayoutGroup, motion } from 'framer-motion'
+import { ReactNode } from 'react'
+import { Info } from 'react-feather'
 
 export type ButtonGroupOption = {
   value: string
   label: string | ReactNode
   disabled?: boolean
+  tabTooltipLabel?: string // Popover tooltip on full tab hover
+  iconTooltipLabel?: string // Popover tooltip on icon hover
+  rightIcon?: ButtonProps['rightIcon']
 }
 
 type Props = {
   currentOption?: ButtonGroupOption
   options: Readonly<ButtonGroupOption[]>
   onChange: (option: ButtonGroupOption) => void
-  size?: ButtonProps['size']
-  width?: string
   groupId: string
+  size?: ButtonProps['size']
+  width?: ButtonProps['width']
+  isFullWidth?: boolean
+  hasLargeTextLabel?: boolean
 }
 
-export default function ButtonGroup({
-  currentOption,
-  options,
-  onChange,
-  size,
-  width,
-  groupId,
-}: Props) {
+export default function ButtonGroup(props: Props) {
+  const { groupId, options, currentOption, isFullWidth } = props
   return (
     <LayoutGroup id={groupId}>
-      <HStack background="level0" p="1" rounded="md" shadow="innerXl" spacing="1">
-        {options.map(option => {
+      <HStack
+        background="level0"
+        p="1"
+        rounded="md"
+        shadow="innerXl"
+        spacing="1"
+        w={isFullWidth ? 'full' : undefined}
+      >
+        {options.map(function (option) {
           const isActive = currentOption?.value === option.value
-          return (
-            <Button
-              bg="transparent"
-              id={`button-group-${option.value}`}
-              isDisabled={option.disabled}
-              key={`button-group-option-${option.value}`}
-              onClick={() => onChange(option)}
-              position="relative"
-              role="group"
-              size={size}
-              variant={isActive ? 'buttonGroupActive' : 'buttonGroupInactive'}
-              width={width}
-            >
-              {isActive && (
-                <Box
-                  as={motion.div}
-                  bg="background.button.secondary"
-                  borderRadius="4px"
-                  inset="0"
-                  layoutId={`active-${groupId}`}
-                  position="absolute"
-                  shadow="md"
-                />
-              )}
-              <Box position="relative" zIndex="8">
-                {option.label}
-              </Box>
-            </Button>
+          return option?.tabTooltipLabel ? (
+            <Box flex="1" key={`button-group-option-${option.value}`}>
+              <Popover trigger="hover">
+                <PopoverTrigger>
+                  <Box _hover={{ opacity: 0.75 }} transition="opacity 0.2s var(--ease-out-cubic)">
+                    <GroupOptionButton isActive={isActive} option={option} {...props} />
+                  </Box>
+                </PopoverTrigger>
+                {option?.tabTooltipLabel && (
```

Comment:
when the label is set correctly this is not needed

Reply:
Removed